File: problems-extend-adapt/3401-3500/3498_weighing-a-string-by-the-mirrored-alphabet/solution.go
// Each character contributes its reversed-alphabet value (26 - letter rank)
// times its 1-indexed string position; sum over the whole string.
func mirroredWeight(s string) int {
	total := 0
	for i := 0; i < len(s); i++ {
		total += (26 - int(s[i]-'a')) * (i + 1)
	}
	return total
}

File: problems-extend-adapt/3401-3500/3498_weighing-a-string-by-the-mirrored-alphabet/solution.rs
impl Solution {
    // Each character contributes its reversed-alphabet value (26 - letter
    // rank) times its 1-indexed string position; sum over the whole string.
    pub fn mirrored_weight(s: String) -> i32 {
        s.bytes()
            .enumerate()
            .map(|(i, b)| (26 - (b - b'a') as i32) * (i as i32 + 1))
            .sum()
    }
}
